src/adhocracy_core/adhocracy_core/sheets/pool.py

```diff
@@ -1,215 +1,123 @@
-from adhocracy_core.utils import append_if_not_none
+from adhocracy_core.interfaces import search_query
+from adhocracy_core.interfaces import SearchQuery
 from adhocracy_core.utils import remove_keys_from_dict
 
 
-dotted_name_resolver = DottedNameResolver()
-
-
-filtering_pool_default_filter = ['depth', 'content_type', 'sheet', 'elements',
-                                 'count', 'sort', 'reverse', 'limit', 'offset',
-                                 'aggregateby', 'aggregateby_elements',
-                                 'only_visible',
-                                 ]
-
+class PoolSheet(AnnotationRessourceSheet):
 
-FilterElementsResult = namedtuple('FilterElementsResult',
-                                  ['elements', 'count', 'aggregateby'])
+    """Pool resource sheet that allows filtering and aggregating elements."""
 
+    _additional_params = ('serialization_form', 'show_frequency', 'show_count')
 
-class PoolSheet(AnnotationStorageSheet):
+    def get(self, params: dict={}, add_back_references=True) -> dict:
+        """Return child references or arbitrary search for descendants.
 
-    """Generic pool resource sheet."""
+        :param params: Parameters to update the search query to find `elements`
+            references (possible items are described in
+            :class:`adhocracy_core.interfaces.SearchQuery`).
+            The default search query is set in the `_references_query`
+            property.
+            If empty only direct child resources are returned and filtered by
+            the target_isheet of the elements field.
 
-    # TODO remove, this class is not used
+        :returns: dictionary with key/values according to
+                  :class:`adhocracy_core.interfaces.SearchResult`
+        """
+        return super().get(params)
 
-    def _get_reference_appstruct(self, params):
-        appstruct = {'elements': []}
-        reftype = self._reference_nodes['elements'].reftype
+    def _get_references_query(self, params: dict) -> SearchQuery:
+        reftype = self._fields['reference']['elements'].reftype
         target_isheet = reftype.getTaggedValue('target_isheet')
-        for child in self.context.values():
-            if target_isheet.providedBy(child):
-                appstruct['elements'].append(child)
+        default_params = {'interfaces': target_isheet,
+                          'root': self.context,
+                          'depth': 1,
+                          'only_visible': False,
+                          'resolve': True,
+                          'allows': (),
+                          }
+        query = search_query._replace(**default_params)
+        if params:
+            query = query._replace(**params)
+        return query
+
+    def _get_reference_appstruct(self, query: SearchQuery) -> dict:
+        if not self._catalogs:
+            return {}  # ease testing
+        result = self._catalogs.search(query)
+        appstruct = {'elements': result.elements,
```

Comment:
I got a `substanced.catalog.CatalogsService` instead of Adhocracy catalog when executing this function via a pyramid script: 

```
def _get_rates_from_user(pool: IPool, user: IUser) -> [IRateVersion]:
    params = {'depth': 6,
              'interfaces': IRate,
              'resolve': True,
              'arbitrary_indexes': {'tag': 'LAST'},
              'references': [(None, IRate, 'subject', user)],
              }
    rates = pool.get(params)['elements']
    return rates
```